app/script/calling/CallingRepository.js

```diff
@@ -112,8 +112,17 @@ z.calling.CallingRepository = class CallingRepository {
    * @returns {undefined} No return value
    */
   shareCallStates() {
-    this.mediaRepository.streamHandler.calls = this.calls;
-    this.mediaRepository.streamHandler.joinedCall = this.joinedCall;
+    this.calls.subscribe(callEntities => {
+      const currentCalls = new Map();
+      callEntities.forEach(callEntity => {
+        const hasPreJoinVideo = callEntity.isIncoming() && callEntity.isRemoteVideoCall();
+        const needsMediaStream = callEntity.selfClientJoined() || hasPreJoinVideo;
+        currentCalls.set(callEntity.id, needsMediaStream && !callEntity.isOngoingOnAnotherClient());
+      });
+      this.mediaRepository.streamHandler.currentCalls = currentCalls;
```

Comment:
Let's convert this to a proper setter instead of a property setting. 
It could ease future debug and has the benefit of throwing errors in case of typos.